chord_metadata_service/chord/export_cbio.py

```diff
@@ -0,0 +1,277 @@
+    lines['genetic_alteration_type'] = 'CLINICAL'
+    if datatype == SAMPLE_DATATYPE:
+        lines['datatype'] = 'SAMPLE_ATTRIBUTES'
+        lines['data_filename'] = SAMPLE_DATA_FILENAME
+    else:
+        lines['datatype'] = 'PATIENT_ATTRIBUTES'
+        lines['data_filename'] = PATIENT_DATA_FILENAME
+
+    for field, value in lines.items():
+        file_handle.write(f"{field}: {value}\n")
+
+
+def individual_export(results, file_handle: TextIO):
+    """
+    Renders Individuals as a clinical_patient text file suitable for
+    importing by cBioPortal.
+
+    cBioPortal Patients fields specs:
+    ---------------------------------
+    Required:
+    - PATIENT_ID
+    Special columns:
+    - OS_STATUS, OS_MONTHS overall survivall. Status can be 1:DECEASED, 0:LIVING
+    - DFS_STATUS, DFS_MONTHS disease free
+    - PATIENT_DISPLAY_NAME
+    - GENDER or SEX
+    - AGE
+    - TUMOR_SITE
+    """
+
+    individuals = []
+    for individual in results:
+        ind_obj = {
+            'id': individual.id,
+            'sex': individual.sex,
+        }
+        individuals.append(ind_obj)
+
+    columns = individuals[0].keys()
+    headers = individual_to_patient_header(columns)
+
+    file_handle.writelines([line + '\n' for line in headers])
+    dict_writer = csv.DictWriter(file_handle, fieldnames=columns, delimiter='\t')
+    dict_writer.writerows(individuals)
+
+
+def sample_export(results, file_handle: TextIO):
+    """
+    Renders Biosamples as a clinical_sample text file suitable for
+    importing by cBioPortal.
+
+    cBioPortal Sample fields specs:
+    ---------------------------------
+    Required:
+    - PATIENT_ID
+    - SAMPLE_ID
+
+    Special columns:
+    - For pan-cancer summary statistics tab:
+        - CANCER_TYPE as an Oncotree code
+        - CANCER_TYPE_DETAILED
+    - SAMPLE_DISPLAY_NAME
+    - SAMPLE_CLASS
+    - METASTATIC_SITE / PRIMARY_SITE overrides the patients level attribute TUMOR_SITE
+    - SAMPLE_TYPE, TUMOR_TISSUE_SITE, TUMOR_TYPE can have the following values
+        (are displayed with a distinct color in the timelines):
+        - "recurrence", "recurred", "progression"
+        - "metastatic", "metastasis"
+        - "primary" or any other value
+    - KNOWN_MOLECULAR_CLASSIFIER
+    - GLEASON_SCORE (prostate cancer)
+    - HISTOLOGY
+    - TUMOR_STAGE_2009
+    - TUMOR_GRADE
+    - ETS_RAF_SPINK1_STATUS
+    - TMPRSS2_ERG_FUSION_STATUS
+    - ERG_FUSION_ACGH
+    - SERUM_PSA
+    - DRIVER_MUTATIONS
+    """
+
+    samples = []
+    for sample in results:
+        # sample.inidividual can be null. Skip the sample in that case.
+        if sample.individual is None:
+            continue
+
+        sample_obj = {
+            'individual_id': sample.individual.id,
+            'id': sample.id
+        }
+        if sample.sampled_tissue:
+            sample_obj['tissue_label'] = sample.sampled_tissue.get('label', '')
+
+        samples.append(sample_obj)
+
+    columns = samples[0].keys()
```

Comment:
In that case, if all our samples are linked via Phenopackets (and for all of them `sample.individual` is None) then our `samples` array will be empty. The line 191 `columns = samples[0].keys()` would give us  `list index out of range` error because the array is empty.
I suggest two options here:
1. Do not skip sample and collect its id
2. Get information about patient id via phenopacket.

Reply:
Done. The code now requests the individual id from phenopacket when it is missing from the `individual` field.
Thanks